controllers/pod.go

```diff
@@ -71,24 +86,39 @@ func (r *SingleClusterReconciler) getRollingRestartTypeMap(
 		}
 
 		podStatus := r.aeroCluster.Status.Pods[pods[idx].Name]
-		if addedNSDevices == nil && podStatus.AerospikeConfigHash != requiredConfHash {
-			// Fetching all block devices that have been added in namespaces.
-			addedNSDevices, err = r.getNSAddedDevices(rackState)
-			if err != nil {
-				return nil, err
+		if podStatus.AerospikeConfigHash != requiredConfHash {
+			if addedNSDevices == nil {
+				// Fetching all block devices that have been added in namespaces.
+				addedNSDevices, err = r.getNSAddedDevices(rackState)
+				if err != nil {
+					return nil, nil, err
+				}
+			}
+
+			// If dynamic commands have failed in previous retry, then we should not try to update config dynamically.
+			if !podStatus.DynamicConfigFailed {
+				// Fetching all dynamic config change.
+				dynamicConfDiffPerPod[pods[idx].Name], err = r.handleDynamicConfigChange(rackState, pods[idx])
+				if err != nil {
+					return nil, nil, err
+				}
```

Comment:
Add a check here or wherever required to not to follow dynamic flow for server versions < 7.0